src/amp-story-player/amp-story-player-impl.js

```diff
@@ -780,6 +781,48 @@ export class AmpStoryPlayer {
     );
   }
 
+  /**
+   * Loads stories in order, giving priority to the story to display first, and
+   * then prerendering the neighbor story once the first one finishes
+   * displaying.
+   * It also builds and positions the iframes accodingly.
+   * @param {!Array<!Object>} loadingQueue
+   * @private
+   */
+  loadStoriesInOrder_(loadingQueue) {
+    const firstStoryIdx = loadingQueue[0].storyIdx;
+    const navigatingBackwards = firstStoryIdx < this.currentIdx_;
+
+    loadingQueue.forEach((currentIframeObj) => {
+      const currentIdx = currentIframeObj.storyIdx;
+      const isVisibleStory = firstStoryIdx === currentIdx;
+      const story = this.stories_[currentIdx];
+      let {iframeIdx} = story;
+      const hasIframe = iframeIdx !== -1;
+
+      // First we must check whether the story has an iframe and allocate one
+      // if it doesn't.
+      if (!hasIframe) {
+        this.allocateIframeForStory_(currentIdx, navigatingBackwards);
+        iframeIdx = story.iframeIdx;
+      }
+
+      // Then we must load the current iframe and neighbor as visible and
+      // prerender respectively.
+      if (currentIframeObj.shouldLoad) {
```

Comment:
With the current implementation, when navigating back you set the same `src` on the `NEXT` story

Reply:
Sorry, not sure I understood. So you mean like calling `show()` and navigating to the previous story?